core/arch/arm/kernel/kern.ld.S

```diff
@@ -356,3 +378,8 @@ SECTIONS
 	}
 
 }
+
+#if defined(CFG_TEE_COHERENT_START)
+ASSERT((__coherent_start & (4096 - 1)) == 0, "Coherent RAM start is not 4Kbyte aligned")
+ASSERT((__coherent_end & (4096 - 1)) == 0, "Coherent RAM end is not 4Kbyte aligned")
```

Comment:
This is extra checks.  Actually `__coherent_end` is already aligned, and static mapping will assert `CFG_TEE_COHERENT_START` (hence `__coherent_start`) is already 4kB page aligned.
Those checks aimed at detecting issues at build time.